core/src/main/java/org/hibernate/ogm/hibernatecore/impl/OgmSessionImpl.java

```diff
@@ -218,6 +224,32 @@ public SimpleNaturalIdLoadAccess bySimpleNaturalId(String entityName) {
 		throw new UnsupportedOperationException( "OGM-589 - Natural id look-ups are not yet supported" );
 	}
 
+	@Override
+	public QueryImplementor createQuery(String queryString) {
+		SessionImpl session = (SessionImpl) delegate;
+		QueryImpl queryImplementor = (QueryImpl) session.createQuery( queryString );
+		OgmQueryImplFacade ogmQueryImplFacade = new OgmQueryImplFacade( queryImplementor );
+
+		return ogmQueryImplFacade;
+	}
+
+	private static class OgmQueryImplFacade<R> extends QueryImpl<R> {
```

Comment:
Maybe you can use the **hint** name in the inner class name.

Reply:
No, Fabio. The inner class is facade of QueryImpl

Reply:
fine